### ISSUE TITLE
Border color not applying with nested ternary

### ISSUE THREAD

**Issue by donkeedong** [vadimdemedes/ink#340]
I'm trying to change a box's border color to reflect the status of an Axios call but it's not working for the "good" state (green). I'm using a nested ternary to decide which state it's in and to pick the color. I've added a `<Text>` component towards the end that uses the same logic for coloring and it works perfectly fine, so I'm not sure what I'm doing wrong. The box correctly changes between blue and red and also correctly changes the border style. I've tried moving the border color ternary out into a separate function and using actually `if` statements with the same results. If I switch the ternary around (like do `isLoading` first or something) then the same thing happens - the last color listed doesn't show up.

```tsx
import axios from "axios";
import { Box, Text } from "ink";
import Spinner from "ink-spinner";
import React, { useEffect, useState } from "react";
import { AppConfig } from "../types/Config";

export const AppBox: React.FC<AppConfig & {
	globalInterval?: number;
}> = props => {
	const [version, setVersion] = useState("???");
	const [isLoading, setIsLoading] = useState(true);
	const [isError, setIsError] = useState(false);

	useEffect(() => {
		let propsInterval = props.interval ? props.interval : props.globalInterval;
		let interval = propsInterval ? propsInterval : 5000;

		const fetchData = () => {
			setIsLoading(true);
			setIsError(false);

			axios
				.get(props.url)
				.then(response => {
					if (props.type === "java") {
						if (
							response.data.build !== undefined &&
							response.data.build.version !== undefined
						) {
							setVersion(response.data.build.version);
						} else {
							setIsError(true);
						}
					} else {
						setVersion(response.data.version);
					}
				})
				.catch(error => {
					if (error.response) {
						setVersion(`Server responded with a ${error.response.status}`);
					} else {
						setVersion(error.message);
					}
					setIsError(true);
				})
				.finally(() => setIsLoading(false));
		};

		const timer = setInterval(fetchData, interval);

		return () => clearInterval(timer);
	}, []);

	return (
		<Box
			padding={1}
			borderStyle={isError ? "double" : "round"}
			borderColor={isError ? "red" : isLoading ? "blue" : "green"}
			flexDirection={"column"}
			width={"50"}
		>
			<Box flexDirection={"row"}>
				<Text color={"lightblue"}>{props.name} </Text>
				<Text color={isError ? "red" : "green"}>
					{isLoading ? `---` : version}{" "}
				</Text>
				<Text color={"yellow"}>
					{isLoading ? <Spinner type={"moon"} /> : null}
				</Text>
			</Box>
			<Text color={isError ? "red" : isLoading ? "blue" : "green"}>
				{isError ? "red" : isLoading ? "blue" : "green"}
			</Text>
		</Box>
	);
};
```

Here's a screenshot of the results:
![image](https://user-images.githubusercontent.com/50036001/88854444-e685c180-d1b6-11ea-98bf-d5877d2922e7.png)

And here's the loading state:
![image](https://user-images.githubusercontent.com/50036001/88854617-29e03000-d1b7-11ea-8d04-f83498977400.png)

This app is just me messing around with ink and showing the status of some of our apps. The app URLs are loaded via a config file on startup and the `<AppBox>` components are dynamically created based on the number of apps. I've tried reordering the apps in the config file with the same results.

**Comment by vadimdemedes** [comment]
Thanks for such a detailed report! This is indeed a bug, found a fix and will push it shortly.

**Comment by vadimdemedes** [comment]
Released a fix in [3.0.1](https://github.com/vadimdemedes/ink/releases/tag/v3.0.1).

**Comment by donkeedong** [comment]
It's working, thanks for the fix!
